packages/yew/src/html/component/mod.rs

```diff
@@ -104,23 +106,34 @@ pub trait BaseComponent: Sized + 'static {
     /// The Component's Properties.
     type Properties: Properties;
 
+    /// The Component's Reference type.
+    type Reference: ErasedStorage;
+
     /// Creates a component.
-    fn create(ctx: &Context<Self>) -> Self;
+    fn create(ctx: &Context<Self>, bindable_ref: BindableRef<Self::Reference>) -> Self;
```

Comment:
Is rebinding allowed? If so, why not move this into `ctx`?

Reply:
At this point, I don't want to allow rebinding it in struct components. Rebinding it in the view method (as in function components) should be fine, but I haven't thought about the other lifecycles. I wouldn't want to change the referenced value unexpectedly for other components, so I haven't thought about any lifecycle methods other than view.

It should be bound by the time `rendered` is run and should continue to exist until the next `view`. I'm not sure if it should be allowed (and which value to expect), if one gets the ref between `view` and `rendered` (in case the value is allowed to rebind).

Struct components can call `BindableRef::forward` to store a ref to forward to the `html!` returned from there `view` method. The same thing has not yet been implemented in function components, but the argument is accessible in `FunctionProvider`. Can be done in a follow up PR though.